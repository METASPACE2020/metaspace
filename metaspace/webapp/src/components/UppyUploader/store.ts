// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
