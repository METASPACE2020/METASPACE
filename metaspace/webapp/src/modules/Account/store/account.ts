import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
