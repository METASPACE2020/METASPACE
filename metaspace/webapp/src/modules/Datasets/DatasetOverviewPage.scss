
.dop {
  @apply grid grid-cols-1 h-full items-start;
  grid-template-areas:
    "dop--header"
    "dop--gallery"
    "dop--details";

  @screen lg {
    @apply grid-cols-2;
    grid-template-areas:
      "dop--header  dop--gallery"
      "dop--details dop--gallery";
  }
}
.dop--header {
  grid-area: dop--header;
}
.dop--details {
  grid-area: dop--details;
}
.dop--gallery {
  grid-area: dop--gallery;
  background: #eeeeee;
  @screen lg {
    height: 100%;
  }
}

.dop--cog-menu {
  @apply flex-none inline-block p-2 m-2 self-center cursor-pointer rounded-full;
  .dop--cog-menu--icon {
    @apply text-3xl;
  }

  &:hover {
    @apply bg-gray-200;
    .dop--cog-menu--icon {
      @apply text-gray-900;
    }
  }
}
