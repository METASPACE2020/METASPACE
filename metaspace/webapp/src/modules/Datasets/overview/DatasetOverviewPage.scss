$className: 'dataset-overview';
$img-viewer-bg-color: #F6F6F6;
$header-height: 64px;
$opt-desc-color: #C4C4C4;


.dataset-overview-container{
  @apply flex flex-row flex-wrap p-0;
  min-height: calc(100vh - #{$header-height});
}

.dataset-overview-img-wrapper{
  background: $img-viewer-bg-color;
}


.dataset-overview-holder{
  @apply flex flex-col justify-center p-6;
}


.dataset-overview-header{
  @apply  flex flex-row items-center justify-between px-6 pt-6;
}



.dataset-opt-description{
  @apply flex flex-row flex-wrap items-center border-solid justify-center border rounded p-2 my-4;
  border-color: #{$opt-desc-color};
  color: #{$opt-desc-color};
}


.dataset-overview-menu{
  z-index: 2000 !important;
}
