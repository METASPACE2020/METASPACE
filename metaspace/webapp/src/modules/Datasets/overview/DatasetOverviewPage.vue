<script src="./DatasetOverviewPage.tsx" />
<style src="./DatasetOverviewPage.scss" lang="scss" />
