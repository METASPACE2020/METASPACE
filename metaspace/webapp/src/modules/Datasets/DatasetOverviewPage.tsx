import { computed, createComponent, ref } from '@vue/composition-api'
import { useQuery } from '@vue/apollo-composable'
import { GetDatasetByIdQuery, getDatasetByIdQuery } from '../../api/dataset'
import { Dropdown, DropdownItem, DropdownMenu } from 'element-ui'
import './DatasetOverviewPage.scss'

export default createComponent<{}>({
  setup(props, ctx) {

    const datasetId = computed(() => ctx.root.$route.params.datasetId)
    const {
      result: datasetResult,
      loading: datasetLoading,
    } = useQuery<GetDatasetByIdQuery>(getDatasetByIdQuery, {id: datasetId})
    const dataset = computed(() => datasetResult.value != null ? datasetResult.value.dataset : null)

    return () => dataset.value != null
      ? (
        <div class="dop--grid">
          <div class="dop--header">
            <div class="flex">
              <h1 class="flex-grow">{dataset.value != null ? dataset.value.name : null}</h1>
              <Dropdown class="dop--cog-menu">
                <i class="el-icon-s-tools dop--cog-menu--icon" />
                <DropdownMenu slot="dropdown">
                  <DropdownItem command="edit">Edit</DropdownItem>
                  <DropdownItem command="editOpt">Upload optical image</DropdownItem>
                  <DropdownItem command="download">Download</DropdownItem>
                  <DropdownItem command="reprocess">Reprocess</DropdownItem>
                  <DropdownItem command="delete">Delete</DropdownItem>
                </DropdownMenu>
              </Dropdown>
            </div>
          </div>
          <div class="dop--details">
            Details
          </div>
          <div class="dop--gallery">
            Gallery
          </div>
        </div>
      )
      : (
        <div class="loading">Loading</div>
      )
  },
})
