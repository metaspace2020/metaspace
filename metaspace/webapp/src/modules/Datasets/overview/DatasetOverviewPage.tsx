import { computed, defineComponent, ref } from 'vue'
import { useQuery } from '@vue/apollo-composable'
import { GetDatasetByIdQuery, getDatasetByIdQuery } from '../../../api/dataset'
import { AnnotationCountTable } from './AnnotationCountTable'
import safeJsonParse from '../../../lib/safeJsonParse'
import { DatasetMetadataViewer } from './DatasetMetadataViewer'
import { DatasetConfigViewer } from './DatasetConfigViewer'
import moment from 'moment'
import { isEmpty } from 'lodash'
import VisibilityBadge from '../common/VisibilityBadge'
import { DatasetActionsDropdown } from './DatasetActionsDropdown'
import { currentUserRoleQuery, CurrentUserRoleResult } from '../../../api/user'
import RichText from '../../../components/RichText'
import isValidTiptapJson from '../../../lib/isValidTiptapJson'
import NewFeatureBadge from '../../../components/NewFeatureBadge'
import './DatasetOverviewPage.scss'
import { useRoute, useRouter } from 'vue-router'
import CopyButton from '../../../components/CopyButton.vue'
// import { RecaptchaV2, useRecaptcha } from 'vue3-recaptcha-v2'

const DatasetOverviewPage = defineComponent({
  name: 'DatasetOverviewPage',
  props: {
    className: {
      type: String,
      default: 'dataset-overview',
    },
    annotationLabel: {
      type: String,
      default: 'Annotations',
    },
    detailLabel: {
      type: String,
      default: 'Details',
    },
    projectLabel: {
      type: String,
      default: 'Projects',
    },
    inpFdrLvls: {
      type: Array,
      default: () => [5, 10, 20, 50],
    },
  },
  // Last reprocessed date (as currently)/Upload date/Number of annotations for FDR 10%/User name/Dataset name
  setup(props) {
    const router = useRouter()
    const route = useRoute()
    // const { handleExecute } = useRecaptcha()
    // @ts-ignore
    const recaptchaToken = ref(process.env.NODE_ENV === 'test' ? 'fake-recaptcha-token' : '')

    const datasetId = computed(() => route.params.dataset_id)
    const { result: datasetResult, loading: datasetLoading } = useQuery<GetDatasetByIdQuery>(getDatasetByIdQuery, {
      id: datasetId,
      inpFdrLvls: props.inpFdrLvls,
    })
    const dataset = computed(() => (datasetResult.value != null ? datasetResult.value.dataset : null))
    const { result: currentUserResult, loading: userLoading } = useQuery<CurrentUserRoleResult | any>(
      currentUserRoleQuery
    )
    const currentUser = computed(() => (currentUserResult.value != null ? currentUserResult.value.currentUser : null))

    const projectLink = (projectIdOrSlug: string) => {
      return {
        name: 'project',
        params: { projectIdOrSlug },
      }
    }

    return () => {
      const {
        name,
        submitter,
        group,
        projects,
        annotationCounts,
        metadataJson,
        id,
        isPublic,
        description,
        canEdit,
        configJson,
        acquisitionGeometry,
        sizeHash,
      } = dataset?.value || ({} as any)
      const { annotationLabel, detailLabel, projectLabel, inpFdrLvls } = props
      const metadata = safeJsonParse(metadataJson) || {}
      const config = safeJsonParse(configJson) || {}
      const acqGeo = safeJsonParse(acquisitionGeometry) || {}
      const fileSize = safeJsonParse(sizeHash) || {}

      // hide deprecated fields
      // eslint-disable-next-line camelcase
      delete metadata?.Submitted_By
      // eslint-disable-next-line camelcase
      delete metadata?.Additional_Information

      const groupLink = group?.id
        ? router.resolve({ name: 'group', params: { groupIdOrSlug: group?.id || '' } }).href
        : ''
      const upDate = moment(moment(dataset?.value?.uploadDT)).isValid()
        ? moment(dataset?.value?.uploadDT).format('D MMMM, YYYY')
        : ''
      const publicationStatus = computed(() => {
        if (Array.isArray(projects) && projects.some(({ publicationStatus }) => publicationStatus === 'PUBLISHED')) {
          return 'Published'
        }
        if (Array.isArray(projects) && projects.some(({ publicationStatus }) => publicationStatus === 'UNDER_REVIEW')) {
          return 'Under review'
        }
        return null
      })
      const dsDescription = isValidTiptapJson(safeJsonParse(description)) ? safeJsonParse(description) : null
      const canViewPublicationStatus =
        dataset.value?.status === 'FINISHED' && canEdit && publicationStatus?.value != null
      // const handleWidgetId = (widgetId: number) => {
      //   handleExecute(widgetId)
      // }
      //
      // const handleLoadCallback = (response: any) => {
      //   recaptchaToken.value = response
      // }

      if ((datasetLoading.value && dataset.value === null) || userLoading.value) {
        return <div class="text-center">Loading...</div>
      } else if (dataset.value === null) {
        return <div class="text-center">This dataset doesn't exist, or you do not have access to it.</div>
      }

      return (
        <div class={'dataset-overview-container justify-center'}>
          <div class={'dataset-overview-wrapper w-full'}>
            <div class="dataset-overview-header">
              <div class="flex flex-row items-center justify-between">
                <h1 class={`truncate ${name.length > 40 ? '!text-2xl' : ''}`}>
                  <span class="break-all whitespace-normal overflow-hidden">{name}</span>
                </h1>
                <CopyButton text={name}>Copy datasetname to clipboard</CopyButton>
                <span class="text-base align-middle">
                  {!isPublic && <VisibilityBadge datasetId={id ? id.toString() : ''} />}
                </span>
              </div>

              <NewFeatureBadge featureKey="imzmlBrowser">
                <DatasetActionsDropdown
                  dataset={dataset?.value}
                  currentUser={currentUser?.value}
                  recaptchaToken={recaptchaToken?.value}
                />
              </NewFeatureBadge>
            </div>
            <div class="dataset-overview-holder">
              <p class="truncate">
                {submitter?.name}
                {group && (
                  <a class="ml-1" href={groupLink}>
                    ({group?.shortName})
                  </a>
                )}
              </p>
              <div>{upDate}</div>
              {description && (
                <RichText class="dataset-opt-description p-0" placeholder=" " content={dsDescription} readonly={true} />
              )}
            </div>
            <div class="dataset-overview-holder">
              <h1 class="truncate">{annotationLabel}</h1>
              <AnnotationCountTable id={id} data={annotationCounts} header={inpFdrLvls} />
            </div>
            {!isEmpty(metadata) && (
              <div class="dataset-overview-holder">
                <h1 class="truncate">{detailLabel}</h1>
                <DatasetMetadataViewer metadata={metadata} />
              </div>
            )}
            {(!isEmpty(config) || !isEmpty(acqGeo)) && (
              <div class="dataset-overview-holder">
                <DatasetConfigViewer data={config} acqGeo={acqGeo} fileSize={fileSize} />
              </div>
            )}
            {Array.isArray(projects) && projects.length > 0 && (
              <div class="dataset-overview-holder">
                <h1 class="truncate">{projectLabel}</h1>
                {projects.map((project) => {
                  return (
                    <div key={project.id} class="flex-grow box-border min-w-64 p-0 break-words">
                      <ul class="list-none p-0 py-3 m-0 max-h-40 overflow-y-auto">
                        <li>
                          <b>Name: </b>
                          <router-link class="ml-1" to={projectLink(project.id)}>
                            {project.name}
                          </router-link>
                        </li>
                        {canViewPublicationStatus && (
                          <li>
                            <b>Status: </b>
                            {publicationStatus?.value}
                          </li>
                        )}
                      </ul>
                    </div>
                  )
                })}
              </div>
            )}
          </div>
        </div>
      )
    }
  },
})

export default DatasetOverviewPage
