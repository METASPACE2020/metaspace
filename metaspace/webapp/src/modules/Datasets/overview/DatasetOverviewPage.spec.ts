import DatasetOverviewPage from './DatasetOverviewPage'
import { nextTick, ref, h, defineComponent } from 'vue'
import { mount, flushPromises } from '@vue/test-utils'
import store from '../../../store'
import router from '../../../router'
import { initMockGraphqlClient } from '../../../tests/utils/mockGraphqlClient'
import { DefaultApolloClient, useQuery } from '@vue/apollo-composable'

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(),
  DefaultApolloClient: vi.fn(),
}))
vi.mock('./DatasetActionsDropdown', () => ({ default: vi.fn() }))
vi.mock('../../../components/NewFeatureBadge', () => ({ default: vi.fn() }))

let graphqlMocks
describe('DatasetOverviewPage', () => {
  const mockDataset = {
    id: 'dataset1',
    name: 'JD_Sampe',
    submitter: {
      id: 'userid',
      name: 'John Doe',
      email: 'jdoe@test.com',
    },
    principalInvestigator: {
      name: 'Test',
      email: 'jdoe@test.com',
    },
    group: {
      id: 'gxxxx',
      name: 'TEST_GROUP',
      shortName: 'TG',
    },
    groupApproved: true,
    projects: [
      {
        id: 'pj1',
        name: 'Test',
        publicationStatus: 'UNPUBLISHED',
      },
    ],
    isPublic: true,
    status: 'FINISHED',
    statusUpdateDT: '2021-03-22T01:54:04.856Z',
    metadataType: 'Imaging MS',
    canEdit: true,
    canDelete: true,
    canDownload: true,
    uploadDT: '2021-03-11T14:29:21.641Z',
    metadataJson: JSON.stringify({
      Metadata_Type: 'Imaging MS',
      MS_Analysis: {
        Detector_Resolving_Power: { mz: 1234, Resolving_Power: 123456 },
      },
    }),
    configJson: JSON.stringify({
      database_ids: [1],
      analysis_version: 3,
      isotope_generation: {
        adducts: ['+H'],
        charge: 1,
        isocalc_sigma: 0.001238,
        instrument: 'Orbitrap',
        n_peaks: 4,
        neutral_losses: [],
        chem_mods: [],
      },
      fdr: {
        decoy_sample_size: 20,
        scoring_model_id: 1,
        model_type: 'catboost',
      },
      image_generation: {
        ppm: 3,
        n_levels: 30,
        min_px: 1,
        compute_unused_metrics: false,
      },
    }),
    description: JSON.stringify({
      type: 'doc',
      content: [
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'Dataset 1.',
            },
          ],
        },
      ],
    }),
  }

  const testHarness = defineComponent({
    components: {
      DatasetOverviewPage,
    },
    setup(props, { attrs }) {
      return () => h(DatasetOverviewPage, { ...attrs, ...props })
    },
  })

  const mockGraphql = async (qyeryParams) => {
    graphqlMocks = await initMockGraphqlClient({
      Query: () => qyeryParams,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(qyeryParams).reduce((acc, key) => ({ ...acc, [key]: qyeryParams[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }

  const noDatasetQuery = async () => {
    await mockGraphql({
      dataset: () => null,
      currentUser: () => ({ id: 'userid', role: 'user' }),
    })
  }
  const overviewQuery = async () => {
    await mockGraphql({
      dataset: () => mockDataset,
      currentUser: () => ({ id: 'userid', role: 'user' }),
    })
  }

  beforeAll(async () => {
    await router.replace({
      name: 'dataset-overview',
      params: {
        dataset_id: 'dataset1',
      },
    })
  })

  it('should match snapshot when dataset exist', async () => {
    await overviewQuery()
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })

    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when dataset does not exist', async () => {
    await noDatasetQuery()
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })

    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
