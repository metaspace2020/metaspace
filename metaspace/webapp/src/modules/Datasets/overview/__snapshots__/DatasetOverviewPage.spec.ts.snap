// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatasetOverviewPage > should match snapshot when dataset does not exist 1`] = `"<div class=\\"text-center\\">This dataset doesn't exist, or you do not have access to it.</div>"`;

exports[`DatasetOverviewPage > should match snapshot when dataset exist 1`] = `
"<div class=\\"dataset-overview-container justify-center\\">
  <div class=\\"dataset-overview-wrapper w-full\\">
    <div class=\\"dataset-overview-header\\">
      <div class=\\"flex flex-row items-center justify-between\\">
        <h1 class=\\"truncate \\"><span class=\\"break-all whitespace-normal overflow-hidden\\">JD_Sampe</span></h1>
        <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
          <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy datasetname to clipboard</p>
        </div><span class=\\"text-base align-middle\\"><!----></span>
      </div>
      <!---->
    </div>
    <div class=\\"dataset-overview-holder\\">
      <p class=\\"truncate\\">John Doe<a class=\\"ml-1\\" href=\\"/group/gxxxx\\">(TG)</a></p>
      <div>11 March, 2021</div>
      <section class=\\"sm-RichText dataset-opt-description p-0\\">
        <!---->
        <div>
          <div class=\\"transition-colors ease-in-out duration-300 rounded bg-transparent\\">
            <div contenteditable=\\"false\\" translate=\\"no\\" class=\\"tiptap ProseMirror\\">
              <p data-placeholder=\\" \\" class=\\"sm-RichText-placeholder is-editor-empty\\"><br class=\\"ProseMirror-trailingBreak\\"></p>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div class=\\"dataset-overview-holder\\">
      <h1 class=\\"truncate\\">Annotations</h1>
      <div class=\\"relative text-center\\">
        <h4># of annotations with FDR% &lt;=</h4>
        <div class=\\"el-table--fit el-table--enable-row-hover has-footer el-table el-table--layout-fixed\\" style=\\"width: 100%;\\" data-prefix=\\"el\\">
          <div class=\\"el-table__inner-wrapper\\">
            <div class=\\"hidden-columns\\">
              <div></div>
              <div></div>
              <div></div>
              <div></div>
              <div></div>
            </div>
            <div class=\\"el-table__header-wrapper\\">
              <table class=\\"el-table__header\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                <colgroup>
                  <col name=\\"el-table_1_column_1\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_2\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_3\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_4\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_5\\" width=\\"80\\">
                </colgroup>
                <thead class=\\"\\">
                  <tr class=\\"\\">
                    <th class=\\"el-table_1_column_1 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                      <div class=\\"cell\\">Database<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                        <!---->
                      </div>
                    </th>
                    <th class=\\"el-table_1_column_2 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                      <div class=\\"cell\\">5%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                        <!---->
                      </div>
                    </th>
                    <th class=\\"el-table_1_column_3 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                      <div class=\\"cell\\">10%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                        <!---->
                      </div>
                    </th>
                    <th class=\\"el-table_1_column_4 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                      <div class=\\"cell\\">20%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                        <!---->
                      </div>
                    </th>
                    <th class=\\"el-table_1_column_5 is-leaf is-sortable el-table__cell\\" colspan=\\"1\\" rowspan=\\"1\\">
                      <div class=\\"cell\\">50%<span class=\\"caret-wrapper\\"><i class=\\"sort-caret ascending\\"></i><i class=\\"sort-caret descending\\"></i></span>
                        <!---->
                      </div>
                    </th>
                  </tr>
                </thead>
              </table>
            </div>
            <div class=\\"el-table__body-wrapper\\">
              <div class=\\"el-scrollbar\\">
                <div class=\\"el-scrollbar__wrap el-scrollbar__wrap--hidden-default\\">
                  <div class=\\"el-scrollbar__view\\" style=\\"display: inline-block; vertical-align: middle;\\">
                    <table class=\\"el-table__body\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\" style=\\"table-layout: fixed;\\">
                      <colgroup>
                        <col name=\\"el-table_1_column_1\\" width=\\"80\\">
                        <col name=\\"el-table_1_column_2\\" width=\\"80\\">
                        <col name=\\"el-table_1_column_3\\" width=\\"80\\">
                        <col name=\\"el-table_1_column_4\\" width=\\"80\\">
                        <col name=\\"el-table_1_column_5\\" width=\\"80\\">
                      </colgroup>
                      <!--v-if-->
                      <tbody tabindex=\\"-1\\"></tbody>
                      <!--v-if-->
                    </table>
                    <div style=\\"height: 100%;\\" class=\\"el-table__empty-block\\"><span class=\\"el-table__empty-text\\">No Data</span></div>
                    <!--v-if-->
                  </div>
                </div>
                <div class=\\"el-scrollbar__bar is-horizontal\\" style=\\"display: none;\\">
                  <div class=\\"el-scrollbar__thumb\\" style=\\"transform: translateX(0%);\\"></div>
                </div>
                <div class=\\"el-scrollbar__bar is-vertical\\" style=\\"display: none;\\">
                  <div class=\\"el-scrollbar__thumb\\" style=\\"transform: translateY(0%);\\"></div>
                </div>
              </div>
            </div>
            <div class=\\"el-table__footer-wrapper\\" style=\\"display: none;\\">
              <table class=\\"el-table__footer\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\">
                <colgroup>
                  <col name=\\"el-table_1_column_1\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_2\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_3\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_4\\" width=\\"80\\">
                  <col name=\\"el-table_1_column_5\\" width=\\"80\\">
                </colgroup>
                <tfoot>
                  <tr>
                    <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_1 is-leaf\\">
                      <div class=\\"cell\\">Total Annotations</div>
                    </td>
                    <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_2 is-leaf\\">
                      <div class=\\"cell\\">
                        <div from=\\"dataset-overview\\">0</div>
                      </div>
                    </td>
                    <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_3 is-leaf\\">
                      <div class=\\"cell\\">
                        <div from=\\"dataset-overview\\">0</div>
                      </div>
                    </td>
                    <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_4 is-leaf\\">
                      <div class=\\"cell\\">
                        <div from=\\"dataset-overview\\">0</div>
                      </div>
                    </td>
                    <td colspan=\\"1\\" rowspan=\\"1\\" class=\\"el-table__cell el-table_1_column_5 is-leaf\\">
                      <div class=\\"cell\\">
                        <div from=\\"dataset-overview\\">0</div>
                      </div>
                    </td>
                  </tr>
                </tfoot>
              </table>
            </div>
            <!--v-if-->
          </div>
          <div class=\\"el-table__column-resize-proxy\\" style=\\"display: none;\\"></div>
        </div>
        <div class=\\"text-right mt-2\\"><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button\\">
            <!--v-if--><span class=\\"\\">Browse annotations</span>
          </button></div>
      </div>
    </div>
    <div class=\\"dataset-overview-holder\\">
      <h1 class=\\"truncate\\">Details</h1>
      <div class=\\"flex flex-wrap relative -m-3\\">
        <div class=\\"flex-grow box-border min-w-64 p-3 break-words\\">
          <!---->
          <p class=\\"max-h-32 overflow-y-auto\\">Imaging MS</p>
        </div>
        <div class=\\"flex-grow box-border min-w-64 p-3 break-words\\">
          <h3 class=\\"m-0 my-2\\">MS analysis</h3>
          <ul class=\\"list-none p-0 m-0 max-h-40 overflow-y-auto\\"></ul>
        </div>
      </div>
    </div>
    <div class=\\"dataset-overview-holder\\">
      <div class=\\"flex flex-wrap relative -m-3\\">
        <div class=\\"flex-grow box-border min-w-64 p-3 break-words\\">
          <h3 class=\\"m-0 my-2\\">Annotation settings</h3>
          <ul class=\\"list-none p-0 m-0 max-h-40 overflow-y-auto\\">
            <li><b>m/z tolerance (ppm):</b> 3</li>
            <li><b>Analysis version:</b> METASPACE-ML</li>
            <!---->
            <!---->
            <!---->
          </ul>
        </div>
      </div>
    </div>
    <div class=\\"dataset-overview-holder\\">
      <h1 class=\\"truncate\\">Projects</h1>
      <div class=\\"flex-grow box-border min-w-64 p-0 break-words\\">
        <ul class=\\"list-none p-0 py-3 m-0 max-h-40 overflow-y-auto\\">
          <li><b>Name: </b>
            <div class=\\"ml-1\\">Test</div>
          </li>
          <!---->
        </ul>
      </div>
    </div>
  </div>
</div>"
`;
